src/components/sidecar/use-active-section.ts

```diff
@@ -25,14 +48,16 @@ export function useActiveSection(
       (entries) => {
         let currentY: number
         let scrollTrend: 'down' | 'up'
-        const visibleHeadings = []
 
         entries.forEach((entry) => {
+          const entryId = entry.target.id
           currentY = window.scrollY
           scrollTrend = previousY.current < currentY ? 'down' : 'up'
 
           if (entry.isIntersecting) {
-            visibleHeadings.push(entry.target.id)
+            visibleHeadings.current[entryId] = entry.target
```

Comment:
Storing references to DOM nodes in this way is _potentially_ dangerous, in that react might recreate them at any moment. An alternative might be to query by the ID when we need the element, but maybe the elements are stable enough in our situation.

Reply:
Great point, I'll make this adjustment!